File: app/src/main/java/bkm/com/newsapp/data/network/NewsFirebaseJobService.java
package bkm.com.newsapp.data.network;

import android.os.Build;
import android.support.annotation.RequiresApi;

import com.firebase.jobdispatcher.JobParameters;
import com.firebase.jobdispatcher.JobService;

import bkm.com.newsapp.utilities.InjectorUtils;

@RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
public class NewsFirebaseJobService extends JobService {
    @Override
    public boolean onStartJob(JobParameters jobParameters) {
        NewsNetworkDataSource networkDataSource = InjectorUtils.provideNetworkDataSource(this.getApplicationContext());
        networkDataSource.fetchNews();
        jobFinished(jobParameters, false);
        return true;
    }

    @Override
    public boolean onStopJob(JobParameters job) {
        return true;
    }
}

File: app/src/main/java/bkm/com/newsapp/data/network/NewsFirebaseJobService.kt
package bkm.com.newsapp.data.network

import android.os.Build
import androidx.annotation.RequiresApi
import bkm.com.newsapp.utilities.InjectorUtils
import com.firebase.jobdispatcher.JobParameters
import com.firebase.jobdispatcher.JobService

@RequiresApi(Build.VERSION_CODES.LOLLIPOP)
class NewsFirebaseJobService : JobService() {

    override fun onStopJob(params: JobParameters): Boolean {
        val networkDataSource = InjectorUtils.provideNetworkDataSource(this)
        networkDataSource?.fetchNews()
        jobFinished(params, false)
        return true
    }

    override fun onStartJob(params: JobParameters?): Boolean {
        return false
    }
}